build-tests/heft-example-plugin-02/src/index.ts

```diff
@@ -0,0 +1,20 @@
+import { IHeftPlugin, HeftSession, HeftConfiguration } from '@rushstack/heft';
+import { default as heftExamplePlugin01 } from 'heft-example-plugin-01';
```

Comment:
`default` is deprecated.  

Suppose the person wrote `import TypeScriptPlugin from 'heft-example-plugin-01';` thinking that they were importing a different plugin.  The compiler errors would be very confusing, because they would refer to it as `TypeScriptPlugin` instead of `ExamplePlugin01`.  Using name-based imports solves this problem, which is why `export default` should be deprecated.

As a second example, suppose you refactored by renaming `ExamplePlugin01` to be a different name. There is no way to automatically fix up all the existing imports -- they will compile without errors, continuing to use the old (wrong) name.

Reply:
`default` is the standard way to say "this is a special export." If you're exporting the plugin's type and the plugin instance, we need a way to say the instance is the important export.

Reply:
We came up with the idea that Heft plugin pacakges should include a "manifest" JSON file with metadata about each individual plugin and its entry point. This would enable Heft to inspect/enumerate the plugins exported by an NPM package without having to execute code.  The manifest can include a field the specifies the module path.

There's nothing special about the word `default`.  The special export could be like this:
```ts
export const heftPluginInstance:  IHeftPlugin = new new ExamplePlugin01();
```
...or like this:
```ts
export function createHeftPlugin(): IHeftPlugin {
  return new new ExamplePlugin01();
}
```

Or it could be a non-special name like this:

```ts
export class ExamplePlugin01 implements IHeftPlugin {
}
```
...and then the manifest would specify the export name:

**heft-plugin-manifest.json**
```ts
"plugins": [
  {
    "name": "Example Plugin",
    "modulePath": "./lib/ExamplePlugin.js",
    "exportName": "ExamplePlugin",
    "description": "..."
  }
]
```

Reply:
(this design change should be a separate PR, however)